ext/node/polyfills/http.ts

```diff
@@ -744,9 +719,12 @@ class ClientRequest extends OutgoingMessage {
     if (!path.startsWith("/")) {
       path = "/" + path;
     }
-    return `${protocol}//${auth ? `${auth}@` : ""}${host}${
+    const url = new URL(`${protocol}//${auth ? `${auth}@` : ""}${host}${
       port === 80 ? "" : `:${port}`
-    }${path}${search}${hash}`;
+    }${path}`);
+    url.search = search;
+    url.hash = hash;
+    return url.href;
```

Comment:
Not sure how to test this, but maybe you'll be able to reproduce the failure with search params we saw in GCP sdk